bitnami/prometheus-operator/templates/prometheus/prometheus.yaml

```
@@ -207,6 +207,7 @@ spec:
               name: {{ .Values.prometheus.thanos.objectStorageConfig.secretName }}
               key: {{ .Values.prometheus.thanos.objectStorageConfig.secretKey | default "thanos.yaml" }}
       {{- end }}
+      resources: {{- toYaml .Values.prometheus.thanos.resources | nindent 8 }}
```

Avoid setting it when the parameter is not set:
```suggestion
      {{- if .Values.prometheus.thanos.resources }}
      resources: {{- toYaml .Values.prometheus.thanos.resources | nindent 8 }}
      {{- end }}
```

---

done